src/firstboot/firstboot.c

```diff
@@ -389,22 +393,26 @@ static int prompt_keymap(void) {
                            kmaps, 60, keymap_is_valid, &arg_keymap);
 }
 
-static int process_keymap(void) {
-        const char *etc_vconsoleconf;
+static int process_keymap(int rfd) {
+        _cleanup_free_ char *etc_vconsoleconf = NULL;
         char **keymap;
         int r;
 
-        etc_vconsoleconf = prefix_roota(arg_root, "/etc/vconsole.conf");
-        if (laccess(etc_vconsoleconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming console has been configured.",
-                          etc_vconsoleconf);
+        r = chase_symlinks_at(rfd, "etc/vconsole.conf",
+                              CHASE_AT_RESOLVE_IN_ROOT|CHASE_NONEXISTENT|CHASE_NOFOLLOW,
+                              &etc_vconsoleconf, NULL);
```

Comment:
similar here. pin the dir for the file, and operate relativeto that to test/create/update the file